tests/common.go

```diff
@@ -7728,3 +7781,274 @@ func GetPoolCapacityUsed(poolUUID string) (float64, error) {
 	return poolSizeUsed, nil
 }
 
+func runDataIntegrityValidation(testName string) bool {
+
+	if Inst().N.String() == ssh.DriverName || Inst().N.String() == vsphere.DriverName {
+		if strings.Contains(testName, "{") {
+			i := strings.Index(testName, "{")
+			j := strings.Index(testName, "}")
+
+			testName = testName[i+1 : j]
+		}
+		if strings.Contains(dataIntegrityValidationTests, testName) {
+			return true
+		}
+
+		if strings.Contains(testName, "Longevity") {
+			pc, _, _, _ := runtime.Caller(1)
+			if strings.Contains(dataIntegrityValidationTests, runtime.FuncForPC(pc).Name()) {
+				return true
+			}
+			return false
+		}
+	}
+
+	return false
+}
+
+func ValidateDataIntegrity(contexts *[]*scheduler.Context) error {
+
+	testName := ginkgo.CurrentGinkgoTestDescription().FullTestText
+	if strings.Contains(testName, "{Longevity}") {
+		pc, _, _, _ := runtime.Caller(1)
+		testName = runtime.FuncForPC(pc).Name()
+	}
+
+	if !runDataIntegrityValidation(testName) {
+		return nil
+	}
+
+outer:
+	for _, ctx := range *contexts {
+		appScaleFactor := time.Duration(Inst().GlobalScaleFactor)
+		var timeout time.Duration
+		if ctx.ReadinessTimeout == time.Duration(0) {
+			timeout = appScaleFactor * defaultTimeout
+		} else {
+			timeout = appScaleFactor * ctx.ReadinessTimeout
+		}
+		//Waiting for all the apps in ctx are running
+		err := Inst().S.WaitForRunning(ctx, timeout, defaultRetryInterval)
+		if err != nil {
+			return err
+		}
+		appVolumes, err := Inst().S.GetVolumes(ctx)
+		if err != nil {
+			return err
+		}
+
+		//waiting for volumes replication status should be up before calculating md5sum
+		for _, v := range appVolumes {
+			replicaSets, err := Inst().V.GetReplicaSets(v)
+			if err != nil {
+				return err
+			}
+
+			//skipping the validation if volume is repl 2
+			if len(replicaSets) == 1 && len(replicaSets[0].PoolUuids) == 1 {
```

Comment:
avoid the use of `continue` whenever possible.
Instead change the condition to be:
```
      if len(replicaSets) > 1 && len(replicaSets[0].PoolUuids) > 1 {
          ..............
          ..............
      }
```